### PR TITLE
Fix seed for train! test

### PR DESCRIPTION
**Pull request by elisno**
CI checks on [#1](https://github.com/SciML/QuantumNLDiffEq.jl/pull/1) were failing due to rng issues. This PR fixes the seed of the rng to get reproducible results from the tests.

Resolves #3

### CODE REVIEW

**Comment by VarLad** [comment]
Thanks for the PR. 

The issue is in bindings for chainrules_patch in YaoBlocks so I don't really get the PR. I'll look into this again, after YaoBlocks tags a new version. :D

**Comment by elisno** [comment]
Good point 👍, I was running the test suite in an environment that used the master branch of Yao/YaoBlocks.

The issue, which this PR addresses, will still persist after the YaoBlocks-fix has been integrated.

# Setup
Checkout the appropriate version of Yao.

I assume it's the fix from https://github.com/QuantumBFS/Yao.jl/commit/2ffc0e56329d95ef89d06bd4112a4d1e7280e11d

```julia
pkg> add Yao#2ffc0e56329d95ef89d06bd4112a4d1e7280e11d
```

# PR comparison

### Before PR
```julia
pkg> test
     # ...
     Testing Running tests...
Match trained equation: Test Failed at /workspaces/QuantumNLDiffEq.jl/test/runtests.jl:24
  Expression: e[10] ≃ -0.657542
   Evaluated: -0.6849492135584023 ≃ -0.657542
Stacktrace:
 [1] macro expansion
   @ /usr/local/julia/share/julia/stdlib/v1.7/Test/src/Test.jl:445 [inlined]
 [2] macro expansion
   @ /workspaces/QuantumNLDiffEq.jl/test/runtests.jl:24 [inlined]
 [3] macro expansion
   @ /usr/local/julia/share/julia/stdlib/v1.7/Test/src/Test.jl:1283 [inlined]
 [4] top-level scope
   @ /workspaces/QuantumNLDiffEq.jl/test/runtests.jl:9
Match trained equation: Test Failed at /workspaces/QuantumNLDiffEq.jl/test/runtests.jl:25
  Expression: e[14] ≃ 0.134493
   Evaluated: 0.13887695118976273 ≃ 0.134493
Stacktrace:
 [1] macro expansion
   @ /usr/local/julia/share/julia/stdlib/v1.7/Test/src/Test.jl:445 [inlined]
 [2] macro expansion
   @ /workspaces/QuantumNLDiffEq.jl/test/runtests.jl:25 [inlined]
 [3] macro expansion
   @ /usr/local/julia/share/julia/stdlib/v1.7/Test/src/Test.jl:1283 [inlined]
 [4] top-level scope
   @ /workspaces/QuantumNLDiffEq.jl/test/runtests.jl:9
Test Summary:          | Pass  Fail  Total
Match trained equation |    4     2      6
ERROR: LoadError: Some tests did not pass: 4 passed, 2 failed, 0 errored, 0 broken.
in expression starting at /workspaces/QuantumNLDiffEq.jl/test/runtests.jl:7
ERROR: Package QuantumNLDiffEq errored during testing
```

### After PR
```julia
pkg> test
     # ...
     Testing Running tests...
Test Summary:          | Pass  Total
Match trained equation |    6      6
     Testing QuantumNLDiffEq tests passed 
```

File: test/runtests.jl [hunk 1]
```diff
@@ -13,14 +15,14 @@ using Zygote: gradient
...
-	@test e[2] ≃ 0.83986
-	@test e[6] ≃ -0.236077
-	@test e[10] ≃ -0.657542
-	@test e[14] ≃ 0.134493
-	@test e[20] ≃ 0.305012
+	@test e[2] ≃ 0.859335
+	@test e[6] ≃ -0.252876
+	@test e[10] ≃ -0.709632
+	@test e[14] ≃ 0.142091
+	@test e[20] ≃ 0.312655
```
Comment (elisno) [inline test/runtests.jl:26]: 1. Rounding to two digits is not enough. 
2. By setting the seed before calling`dispatch(EasyBuild.variational_circuit(6,5), :random)` and `train!(quantum_feature_map_circuit, cost, var, M, N, 1.0, params)`, I get these current values.

**Comment by VarLad** [comment]
It seems that you're right on this :D 
I need some more time to think on this! :) 

Again, thanks a lot for the PR and the report!

**Comment by VarLad** [comment]
Hey, can you try again with latest commit?

**Comment by elisno** [comment]
I added the latest commits after a new version of Yao was released ([v0.7.2](https://github.com/QuantumBFS/Yao.jl/releases/tag/v0.7.2)). It contains the fixes you mentioned @VarLad.

Note that we're working with a very brittle test case. I recommend fixing the optimizer and hyperparameters that are provided to `train!` when running tests.

In any case, I think this PR is ready to merge with your approval @VarLad .

**Comment by VarLad** [comment]
Just need some time to figure some other things out! :D

**Comment by elisno** [comment]
Closing after update in ec20c908fd48a1eb898ef56d657c30825c0a3b74.

**Pull request closed without merging**